### PR TITLE
Refactor plot contour

### PR DESCRIPTION
**Pull request by not522**
## Motivation

This is a follow up of #5778.

## Description of the changes

This PR simplifies some unnecessary implementation.

### CODE REVIEW

**Comment by nabenabe0928** [comment]
@HideakiImamura 

Could you review this PR?

File: optuna/visualization/matplotlib/_contour.py [hunk 1]
```diff
@@ -261,90 +239,64 @@ def _calculate_griddata(
...
         return None
 
-    (
-        xi,
-        yi,
-        zi,
-        x_cat_param_pos,
-        x_cat_param_label,
-        y_cat_param_pos,
-        y_cat_param_label,
```
Comment (nabenabe0928) [inline optuna/visualization/matplotlib/_contour.py:312]: I confirmed that these return values are now handled by `_calculate_axis_data`.

File: optuna/visualization/matplotlib/_contour.py [hunk 2]
```diff
@@ -261,90 +239,64 @@ def _calculate_griddata(
...
-        feasible_plot_values,
-        infeasible_plot_values,
-    ) = _calculate_griddata(info)
+    zi, feasible_plot_values, infeasible_plot_values = _calculate_griddata(info)
     cs = None
     if len(zi) > 0:
-        if info.xaxis.is_log:
-            ax.set_xscale("log")
-        if info.yaxis.is_log:
-            ax.set_yscale("log")
```
Comment (nabenabe0928) [inline optuna/visualization/matplotlib/_contour.py:321]: These scalings are now done here:
https://github.com/optuna/optuna/blob/2e143f2e778230a8c4ed514967ec63e6232d2092/optuna/visualization/matplotlib/_contour.py#L263
https://github.com/optuna/optuna/blob/2e143f2e778230a8c4ed514967ec63e6232d2092/optuna/visualization/matplotlib/_contour.py#L268

File: optuna/visualization/matplotlib/_contour.py [hunk 3]
```diff
@@ -261,90 +239,64 @@ def _calculate_griddata(
...
-        if info.xaxis.name != info.yaxis.name:
```
Comment (nabenabe0928) [inline optuna/visualization/matplotlib/_contour.py:322]: We know that the names are different thanks to this line.
https://github.com/optuna/optuna/blob/2e143f2e778230a8c4ed514967ec63e6232d2092/optuna/visualization/matplotlib/_contour.py#L270

**Review by nabenabe0928 — APPROVED** [review]
Thank you for the PR, LGTM!
I confirmed all the changes and left the comments why each change is valid.

**Review by HideakiImamura — APPROVED** [review]
Thanks for the follow-up. LGTM.

**Pull request merged**